### PR TITLE
Fix gluon Trainer with AMP in Mask-RCNN training script

### PR DESCRIPTION
**Pull request by Kh4L**
AMP without Horovod was failing.
Set `kvstore` to `None` when using `--amp`.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-959-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-959/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by Jerryzcn** [comment]
can you set it to device?

File: scripts/instance/mask_rcnn/train_mask_rcnn.py [hunk 1]
```diff
@@ -326,7 +326,8 @@ def train(net, train_data, val_data, eval_metric, batch_size, ctx, args):
             net.collect_train_params(),  # fix batchnorm, fix first stage, etc...
             'sgd',
             optimizer_params,
-            update_on_kvstore=(False if args.amp else None), kvstore=kv)
+            update_on_kvstore=(False if args.amp else None),
+            kvstore=(None if args.amp else kv))
```
Comment (Jerryzcn) [inline scripts/instance/mask_rcnn/train_mask_rcnn.py]: I think 
``` kv = mx.kvstore.create(args.kv_store if not args.amp else 'device')```
is better. The root cause seems to be that nccl kvstore does not support amp

Comment (Kh4L) [reply scripts/instance/mask_rcnn/train_mask_rcnn.py]: Done

**Comment by mli** [comment]
Job PR-959-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-959/2/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**